### PR TITLE
Feature/refactoring

### PR DESCRIPTION
**Pull request by stefan-prokop-cz**

### CODE REVIEW

File: src/templates/withDefaultContext.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import * as intl from 'intl';
 import { defaults } from 'lodash';
-import * as moment from 'moment';
+
+// tslint:disable-next-line:no-var-requires
+const { formatToTimeZone } = require('date-fns-timezone');
```
Comment (vlasy) [inline src/templates/withDefaultContext.ts]: Why not import?

File: src/createOffice.ts [hunk 2]
```diff
@@ -46,13 +46,13 @@ export const createOffice = (options: ServiceOptions, ident: string = 'default')
         /* eslint-disable global-require */
```
Comment (grissius) [inline src/createOffice.ts]: Is there still need for this disable?

File: src/offices/GmailOffice.ts [hunk 3]
```diff
@@ -6,33 +6,49 @@ import {ServiceType} from '../createOffice';
...
     mailer?: any;
+} & (
+    | {
+    authType: GmailAuthType.login;
+    settings: GmailLoginOptions;
 }
+    | {
+    authType: GmailAuthType.oauth2;
+    settings: GmailOAuth2Options;
+});
```
Comment (grissius) [inline src/offices/GmailOffice.ts]: :ok_man:

File: src/templates/withDefaultContext.ts (hunk 1 above)
Comment (grissius) [reply src/templates/withDefaultContext.ts]: Agree, types are included in `date-fns-timezone`. No need for disable nor require.

File: src/offices/GmailOffice.ts (hunk 3 above)
Comment (grissius) [reply src/offices/GmailOffice.ts]: Noice

File: src/createOffice.ts (hunk 2 above)
Comment (stefan-prokop-cz) [reply src/createOffice.ts]: no longer .. removed

File: src/templates/withDefaultContext.ts (hunk 1 above)
Comment (stefan-prokop-cz) [reply src/templates/withDefaultContext.ts]: :thinking: i dont know why i used this

fixed

File: README.md [hunk 4]
```diff
@@ -198,6 +121,11 @@ const mail: Mail = {
 };
 ```
 
+## Used packages
```
[the remark this answers is not in the document]
Comment (stefan-prokop-cz) [reply README.md]: Yes its unnecessary. Already removed from readme

File: src/offices/GmailOffice.ts [hunk 5]
```diff
@@ -6,33 +6,49 @@ import {ServiceType} from '../createOffice';
 import { Mail, MailType } from '../Mail';
 import { Office } from './Office';
 
+type GmailAuthOptions = GmailLoginOptions | GmailOAuth2Options;
```
[the remark this answers is not in the document]
Comment (stefan-prokop-cz) [reply src/offices/GmailOffice.ts]: already using nodemailer types and also add `@types/nodemailer` to dependencies

File: src/templates/withDefaultContext.ts [hunk 6]
```diff
@@ -38,11 +38,12 @@ export const withDefaultContext = (context: any = {}) => {
 };
 
 const shiftUtcDate = (d: any, t: any) => {
+    const dateFormat = 'YYYY-MM-DD HH:mm:ss';
     if (!d) {
         return d;
     }
     if (t) {
-        return new Date(moment.utc(d).tz(t).format('YYYY-MM-DD HH:mm:ss'));
+        return new Date(formatToTimeZone(d, dateFormat, { timeZone: t }));
```
[the remark this answers is not in the document]
Comment (stefan-prokop-cz) [reply src/templates/withDefaultContext.ts]: We can remove `utc` --> `const defaultTz = context.tz` or we can replace it via another timezone. I think we can remove default `utc` value.

File: README.md [hunk 7]
```diff
@@ -56,88 +61,6 @@ getOffice()
...
-    filters: [/* ... */],
-    android_sound: 'default',
-}
-```
-
-### Sendgrid
-
-Would be same as `gmail`, you only need to change `ServiceType.gmail` and add your `apiKey`
-
 ## Concept
```
[the remark this answers is not in the document]
Comment (stefan-prokop-cz) [reply README.md]: Sendgrid readme changed - usage added

File: src/templates/withDefaultContext.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (smoliji) [reply src/templates/withDefaultContext.ts]: please do :pray: all i want is to support common `utc` timezone string.

or add a breaking change to changelog and this must come in next major release.

[the remark this answers is not in the document]
Comment (stefan-prokop-cz) [reply src/templates/withDefaultContext.ts]: `utc` support added

**Pull request merged**